packages/keyring-controller/src/KeyringController.ts

```diff
@@ -807,7 +875,7 @@ export class KeyringController extends BaseControllerV2<
 
     keyring.setAccountToUnlock(index);
     const oldAccounts = await this.#keyring.getAccounts();
-    await this.#keyring.addNewAccount(keyring);
+    await this.#keyring.addNewAccount(keyring as unknown as Keyring<Json>);
```

Comment:
What do you think about adding a comment explaining why this is being typecasted?

Reply:
Added!